### PR TITLE
ghwatch: Auto-update metadata repositories

### LINKED ISSUE #79: ghwatch: Auto-update metadata repositories
ghwatch can make use of a clone of `bitcoin-gh-meta` (and potentially others) to have issue and PR details available locally, it would be useful if it could (optionally) periodically `git pull` these in the background to keep them up to date.

### PR DESCRIPTION
**Pull request by Kvaciral**
Closes #79

### CODE REVIEW

File: ghwatch.py [hunk 1]
```diff
@@ -310,19 +313,32 @@ def main():
 
     Key.start(hide_cursor=True)
 
-    t = 0
+    refr_t = 0
+    meta_t = 0
+    meta_i = config.get('auto-update', 0)
```
Comment (laanwj) [inline ghwatch.py:318]: Please call this variable `auto_update` or something else clearer (using `t` for the counter is fine).

**Comment by laanwj** [comment]
Thanks ! Changes look good to me overall, will test.

File: ghwatch.py (hunk 1 above)
Comment (Kvaciral) [reply ghwatch.py]: Done!! :D

File: README.md [hunk 2]
```diff
@@ -348,13 +348,15 @@ Then, edit the configuration file. Only thing that is necessary to change is `gh
...
 
-If you want to see PR status (and other issue details like labels), point `meta` for the `bitcoin/bitcoin` repository to an up-to-date checkout of [bitcoin-gh-meta](https://github.com/zw/bitcoin-gh-meta). One way to keep this repository up to date is with a crontab entry that periodically does `git pull`.
+If you want to see PR status (and other issue details like labels), point `meta` for the `bitcoin/bitcoin` repository to an up-to-date checkout of [bitcoin-gh-meta](https://github.com/zw/bitcoin-gh-meta).
 ```
     "meta": {
         "bitcoin/bitcoin": "/path/to/bitcoin-gh-meta"
     },
 ```
 
+To keep this repository up to date you can set the interval in seconds in 'auto-update', default is 0 (i.e. no auto-update).
```
Comment (jonatack) [inline README.md:358]: naming nit, perhaps use snakecase like `label_prio` for `auto_update`
```suggestion
To keep this repository up to date you can set the interval in seconds in 'auto_update', default is 0 (i.e. no auto-update).
```

**Review by jonatack — COMMENTED** [review]
Nice!

File: ghwatch.py [hunk 3]
```diff
@@ -33,6 +33,9 @@
     # Repository with github metadata mirror (to get label data)
     'meta': {'bitcoin/bitcoin': '/path/to/bitcoin-gh-meta'},
 
+    # Interval in seconds for an auto-update (git pull) of github metadata mirror if greater than 0.
```
Comment (jonatack) [inline ghwatch.py:36]:
```suggestion
    # Interval in seconds for an auto-update (git pull) of github metadata mirror, if greater than 0.
```

File: README.md (hunk 2 above)
Comment (jonatack) [reply README.md:358]: It looks like bitcoin-gh-meta performs a refresh every 2 hours. Maybe mention that in the `auto_update` documentation. After the first auto-update git pull, settings under 2 hours won't provide any benefit.

Perhaps a refresh could be done on launch if auto_update > 0.

File: ghwatch.py [hunk 4]
```diff
@@ -33,6 +33,9 @@
...
+    'auto-update': 0,
```
Comment (vmedea) [inline ghwatch.py:37]: Might want to make this `auto_update`—no strong opinion on snake case versus kebab case but it's good to be consistent with `label_prio`.

File: README.md (hunk 2 above)
Comment (Kvaciral) [reply README.md]: Done, and done.. thanks for the suggestions!!

**Comment by laanwj** [comment]
Tested ACK 89d0b3c91dc2031dc03f80155a920c1be8251b92
Both the initial update and the periodic update are working great here.

File: ghwatch.py [hunk 5]
```diff
@@ -305,28 +320,37 @@ def main():
...
             k = Key.get()
             if not k:
                 Key.input_wait(1.0)
-                t += 1
+                refr_t += 1
+                meta_t += 1
                 continue
             if k == 'escape':
                 break
-            # print(k, Key.mouse_pos)
```
Comment (jonatack) [inline ghwatch.py:329]: Was this comment inadvertently removed?

**Comment by jonatack** [comment]
ACK, tested (including with internet off to check the unhappy path) and seems to be working well here too.

File: ghwatch.py (hunk 5 above)
Comment (laanwj) [reply ghwatch.py:329]: It's not strictly part of this change but imo it's okay to remove it,
After all it is not a useful comment only a commented-out debug statement I accidentally left around :slightly_smiling_face:

**Comment by jonatack** [comment]
ACK 89d0b3c91dc2031dc03f80155a920c1be8251b92, tested (including with internet off to check the unhappy paths using `auto_update` with 0, more than 0, and absent) and seems to be working well here too. Thanks!

**Pull request merged**